test/github_app.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+set -e
+
+source "$(dirname "$0")/helpers.sh"
+if [ -d /opt/resource ]; then
+  source /opt/resource/common.sh
+else
+  source "$(dirname "$0")/../assets/common.sh"
+fi
+
+# Test case 1: setup_github_app_credentials without GitHub App credentials
+it_skips_setup_when_github_app_credentials_not_provided() {
+    export payload='{
+        "source": {
+            "uri": "https://github.com/test/repo.git"
+        }
+    }'
+    export uri="https://github.com/test/repo.git"
+
+    # This should not error and should not modify git config
+    setup_github_app_credentials
+
+    # Verify that git config was not modified with credential helper
+    ! git config --global credential.helper | grep -q "x-access-token" || true
+}
+
+# Test case 2: setup_github_app_credentials with incomplete credentials (missing private key)
+it_skips_setup_when_github_app_credentials_incomplete() {
+    export payload='{
+        "source": {
+            "uri": "https://github.com/test/repo.git",
+            "github_app_id": "123456"
+        }
+    }'
+    export uri="https://github.com/test/repo.git"
+
+    # This should not error and should not modify git config
+    setup_github_app_credentials
+
+    # Verify that git config was not modified
+    ! git config --global credential.helper | grep -q "x-access-token" || true
+}
+
+# Test case 3: setup_github_app_credentials extracts credentials correctly
+it_extracts_github_app_credentials() {
+    local test_key="test-private-key-content"
+    export payload='{
+        "source": {
+            "uri": "https://github.com/test/repo.git",
+            "github_app_id": "123456",
+            "github_app_private_key": "'"$test_key"'",
+            "github_app_installation_id": "789012"
+        }
+    }'
+    export uri="https://github.com/test/repo.git"
+
+    # Extract the values to verify they are extracted correctly
+    github_app_id=$(jq -r '.source.github_app_id // ""' <<< "$payload")
+    github_app_private_key=$(jq -r '.source.github_app_private_key // ""' <<< "$payload")
+    github_app_installation_id=$(jq -r '.source.github_app_installation_id // ""' <<< "$payload")
+
+    test "$github_app_id" = "123456"
+    test "$github_app_installation_id" = "789012"
+    test "$github_app_private_key" = "$test_key"
+}
+
+# Test case 4: setup_github_app_credentials converts SSH URI to HTTPS
+it_converts_ssh_uri_to_https() {
+    export uri="git@github.com:test/repo.git"
+
+    # Simulate the URI conversion logic
+    if [[ $uri == git@github.com:* ]]; then
+        uri="https://github.com/${uri#git@github.com:}"
+    fi
```

Comment:
This doesn't actually test the code that gets used in `common.sh`.

Why not put the code from `common.sh` into a function and then test the function?